Analytical_Engine/Query/Graph/DistinctEntities.cs

```diff
@@ -43,36 +43,27 @@ public static partial class Query
 
         [Description("Identifies unique objects from a collection IBHoMObjects using hash comparison.")]
         [Input("entities", "A collection of IBHoMObjects from which unique instances are identified.")]
-        [Input("diffConfig", "Configuration of diffing used to find unique entities.")]
+        [Input("comparisonConfig", "Configuration of diffing used to find unique entities.")]
         [Output("unique entities", "A Dictionary replacement map of the entities where the keys are the Guid of the original entity and the Values the matching IBHoMObject entity.")]
-        public static Dictionary<Guid, IBHoMObject> DiffEntities(this List<IBHoMObject> entities, DiffConfig diffConfig = null)
+        public static Dictionary<Guid, IBHoMObject> DistinctEntities(this List<IBHoMObject> entities, ComparisonConfig comparisonConfig = null) // this is not a diff, it's finding unique entities
         {
-            if (diffConfig == null)
-                diffConfig = new DiffConfig();
+            ComparisonConfig cc = comparisonConfig ?? new ComparisonConfig();
 
             Dictionary<Guid, IBHoMObject> replaceMap = new Dictionary<Guid, IBHoMObject>();
             Dictionary<IBHoMObject, string> objectHash = new Dictionary<IBHoMObject, string>();
 
-            //remove old hashes if any
-            RemoveExistingHashes(entities);
-
-            //find decimal places from numeric tolerance
-            int decimalPlaces = DecimalPlaces(diffConfig.NumericTolerance);
-
-            //generate hashes
-            entities.ForEach(ent => objectHash.Add(ent, ent.HashEntity(diffConfig, decimalPlaces)));
+            HashComparer<object> hashComparer = new HashComparer<object>(cc);
 
             foreach (KeyValuePair<IBHoMObject, string> entityA in objectHash)
             {
```

Comment:
Probably I am ignorant of how to use the HashComparer. So maybe I missing something but here no comparisons take place.

Reply:
Comparison should happen on line 65 (`if (hashComparer.Equals(entityA.Value, entityB.Value))`) by calling Equals.

Behaviour in my mind should've been the same as before, at least that was my intention when I refactored the code. However I did not test this function (`DistinctEntities`) but I did test the `HashComparer.Equals` and that works.

However I did expect I might have done something wrong here, so happy to set up a call to see together what the problem might be 😄